### PR TITLE
TTL update & remove struct AckInner

### LINKED ISSUE #9: Update TTL type & remove struct AckInner
### TTL
Just wanted to suggest to limit the size of the **ttl** property in all of the structs.
There's no need to have it that big. A `u8` type might be sufficient, it'll save space and it'll align with the network topology and the modern internet standards with a max **TTL** of 255. Also a possible default value could be 64.

### `struct AckInner {}`
In the definition of the Ack struct we're using the AckInner struct like this:
```
pub struct Ack(AckInner);

struct AckInner {
	session_id: u64,
	when: std::time::Instant,
	// Time at which the message was received.
}
```
I think it's redundant and I suggest to remove AckInner and move the attributes in the Ack struct itself. Like this:
```
struct Ack {
	session_id: u64,
	when: std::time::Instant,
	// Time at which the message was received.
}
```

### PR DESCRIPTION
**Pull request by IlyaEmeliyanov**
Resolves #9

### CODE REVIEW

File: AP-protocol.md [hunk 1]
```diff
@@ -436,4 +434,4 @@ Notice that these messages are not subject to the rules of fragmentation, in fac
 - S -> C : client_list!(list_length, list_of_client_ids)
 - C -> S : message_for?(client_id, message_size, message)
 - S -> C : message_from!(client_id, message_size, message)
-- S -> C : error_wrong_client_id!
\ No newline at end of file
+- S -> C : error_wrong_client_id!
```
Comment (matteo-dinoia) [inline AP-protocol.md:437]: Not sure what changed here (maybe you changed the spacing or removed/added \n)

File: AP-protocol.md [hunk 2]
```diff
@@ -287,9 +287,7 @@ Source Routing Header contains the path to the client, which can be obtained by
 If a drone receives a Message and can forward it to the next hop, it also sends an Ack to the client.
 
 ```rust
-pub struct Ack(AckInner);
-
-struct AckInner {
+struct Ack {
```
Comment (matteo-dinoia) [inline AP-protocol.md:290]: This is fine but maybe it should be in another pull request.
At least it should be in another commit.

And same for the pub in the L336

**Review by matteo-dinoia — CHANGES_REQUESTED** [review]
Seem fine by me only fix the problem cited

File: AP-protocol.md (hunk 2 above)
Comment (IlyaEmeliyanov) [reply AP-protocol.md:290]: Ok

**Comment by matteo-dinoia** [comment]
Can you rebase on newer changes (or just create a new PR) with only the u8 stuff? Then I'll merge it if nobody is against it.
